### PR TITLE
Replaced 'assert()' with 'rz_return_if_fail()' #3778

### PR DESCRIPTION
**Pull request by ezinneanne**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

I replaced the `assert()` keyword with `rz_return_if_fail()` in all the lines specified in #3778 excluding `librz/bin/dwarf/op.c` and `librz/core/project_migrate.c` 

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/include/rz_util/rz_assert.h [hunk 1]
```diff
@@ -137,19 +137,19 @@ RZ_API void rz_assert_log(RzLogLevel level, const char *fmt, ...) RZ_PRINTF_CHEC
 
 #define rz_return_if_fail(expr) \
 	do { \
-		assert(expr); \
+		rz_return_if_fail(expr); \
```
Comment (XVilka) [inline librz/include/rz_util/rz_assert.h:140]: In this file this change is unnecessary - it will just make it recursive

File: librz/util/buf_sparse.c [hunk 2]
```diff
@@ -167,7 +167,7 @@ static bool buf_sparse_resize(RzBuffer *b, ut64 newsize) {
 		RzBufferSparseChunk *c = rz_vector_index_ptr(&priv->chunks, n - 1);
 		if (newsize <= c->to) {
 			// must chop the now-last block
-			assert(newsize); // newsize > 0 is guaranteed when n > 0, otherwise the lower bound above would have returned 0.
+			rz_return_if_fail(newsize); // newsize > 0 is guaranteed when n > 0, otherwise the lower bound above would have returned 0.
```
Comment (XVilka) [inline librz/util/buf_sparse.c:170]: As this function returns a value, `rz_return_val_if_fail()` should be used, e.g. `rz_return_val_if_fail(newsize, false)`
The same applies to other spaces where function returns value

File: librz/asm/arch/include/mybfd.h [hunk 3]
```diff
@@ -4535,15 +4535,15 @@ bfd_boolean generic_core_file_matches_executable_p(bfd *core_bfd, bfd *exec_bfd)
 #ifdef DEBUG_BFD_SEND
 #undef BFD_SEND
 #define BFD_SEND(bfd, message, arglist) \
-	(((bfd) && (bfd)->xvec && (bfd)->xvec->message) ? ((*((bfd)->xvec->message))arglist) : (bfd_assert(__FILE__, __LINE__), NULL))
+	(((bfd) && (bfd)->xvec && (bfd)->xvec->message) ? ((*((bfd)->xvec->message))arglist) : (bfd_rz_return_if_fail(__FILE__, __LINE__), NULL))
```
Comment (XVilka) [inline librz/asm/arch/include/mybfd.h:4538]: This seems to be a result of the `sed`, unnecessary change

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/asm/arch/include/mybfd.h [hunk 4]
```diff
@@ -4535,15 +4535,15 @@ bfd_boolean generic_core_file_matches_executable_p(bfd *core_bfd, bfd *exec_bfd)
...
 #endif
 #define BFD_SEND_FMT(bfd, message, arglist) \
 	(((bfd)->xvec->message[(int)((bfd)->format)])arglist)
 
 #ifdef DEBUG_BFD_SEND
 #undef BFD_SEND_FMT
 #define BFD_SEND_FMT(bfd, message, arglist) \
-	(((bfd) && (bfd)->xvec && (bfd)->xvec->message) ? (((bfd)->xvec->message[(int)((bfd)->format)])arglist) : (bfd_assert(__FILE__, __LINE__), NULL))
+	(((bfd) && (bfd)->xvec && (bfd)->xvec->message) ? (((bfd)->xvec->message[(int)((bfd)->format)])arglist) : (bfd_rz_return_if_fail(__FILE__, __LINE__), NULL))
```
Comment (wargio) [inline librz/asm/arch/include/mybfd.h:4546]: same here

File: librz/include/rz_util/rz_assert.h (hunk 1 above)
Comment (ezinneanne) [reply librz/include/rz_util/rz_assert.h:140]: Okay so I should still use assert instead?

File: librz/util/buf_sparse.c (hunk 2 above)
Comment (ezinneanne) [reply librz/util/buf_sparse.c:170]: Okay thank you

File: librz/asm/arch/include/mybfd.h (hunk 3 above)
Comment (ezinneanne) [reply librz/asm/arch/include/mybfd.h:4538]: Okay so I shouldn't change this?
Sorry I actually got confused. Please is there a way I could know where to place the changes and where not to ?

File: librz/asm/arch/include/mybfd.h (hunk 4 above)
Comment (ezinneanne) [reply librz/asm/arch/include/mybfd.h:4546]: Okay so I shouldn't change this too?

File: librz/include/rz_util/rz_assert.h (hunk 1 above)
Comment (XVilka) [reply librz/include/rz_util/rz_assert.h:140]: In this file, which defines macroses like `rz_return_if_fail()` - yes.

File: librz/asm/arch/include/mybfd.h (hunk 3 above)
Comment (XVilka) [reply librz/asm/arch/include/mybfd.h:4538]: Yes, you could see it's not a pure `assert()` but `bfd_assert()`

File: librz/asm/arch/include/mybfd.h (hunk 4 above)
Comment (wargio) [reply librz/asm/arch/include/mybfd.h:4546]: yes, like above.
